CI: Unit tests (Travis): PANIC in TestSpeakerOnUpdateNode

## CI failure

- Failing test: https://app.travis-ci.com/github/cilium/cilium/builds/253157677 (https://github.com/cilium/cilium/pull/20479)
- Raw log from Travis: [log.txt](https://github.com/cilium/cilium/files/9103641/log.txt)


```
$ ./.travis/build.sh

[...]

  TEST  github.com/cilium/cilium/pkg/bgp/speaker
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   PANIC  package: github.com/cilium/cilium/pkg/bgp/speaker • TestSpeakerOnUpdateNode   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x185e376]

goroutine 48 [running]:
github.com/cilium/cilium/pkg/bgp/mock.(*MockMetalLBSpeaker).GetBGPController(0xc00057aa80?)
        /home/travis/gopath/src/github.com/cilium/cilium/pkg/bgp/mock/mock.go:38 +0x16
github.com/cilium/cilium/pkg/bgp/speaker.(*MetalLBSpeaker).announcePodCIDRs(0xc00036b6e0, {0xc00040bb60, 0x1, 0x1})
        /home/travis/gopath/src/github.com/cilium/cilium/pkg/bgp/speaker/pod_cidr.go:84 +0x131
github.com/cilium/cilium/pkg/bgp/speaker.(*MetalLBSpeaker).handleNodeEvent(0xc00036b6e0, {{{0x0, 0x0}, 0x1}, 0x2, 0xc00011cd40, 0xc00035e3a8, 0x0})
        /home/travis/gopath/src/github.com/cilium/cilium/pkg/bgp/speaker/events.go:231 +0x445
github.com/cilium/cilium/pkg/bgp/speaker.(*MetalLBSpeaker).do(0xc00036b6e0, {0x1b427e0?, 0xc0006e2500?})
        /home/travis/gopath/src/github.com/cilium/cilium/pkg/bgp/speaker/events.go:187 +0x496
github.com/cilium/cilium/pkg/bgp/speaker.(*MetalLBSpeaker).run(0xc00036b6e0, {0x1f5afb8, 0xc00036b620})
        /home/travis/gopath/src/github.com/cilium/cilium/pkg/bgp/speaker/events.go:114 +0x1e9
created by github.com/cilium/cilium/pkg/bgp/speaker.TestSpeakerOnUpdateNode
        /home/travis/gopath/src/github.com/cilium/cilium/pkg/bgp/speaker/speaker_test.go:368 +0x470
FAIL    github.com/cilium/cilium/pkg/bgp/speaker        0.036s

┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  STATUS │ ELAPSED │                          PACKAGE                          │ COVER  │ PASS │ FAIL │ SKIP  │
│─────────┼─────────┼───────────────────────────────────────────────────────────┼────────┼──────┼──────┼───────│
│  PANIC  │ 0.00s   │ github.com/cilium/cilium/pkg/bgp/speaker                  │ --     │ --   │ --   │ --    │
│  PASS   │ 0.01s   │ github.com/cilium/cilium/bugtool/cmd                      │ 12.2%  │    1 │    0 │    0  │
│  PASS   │ 0.05s   │ github.com/cilium/cilium/cilium/cmd                       │ 14.3%  │    1 │    0 │    0  │
│  PASS   │ 37.82s  │ github.com/cilium/cilium/daemon/cmd                       │ 26.3%  │    2 │    0 │    0  ���
│  PASS   │ 0.04s   │ github.com/cilium/cilium/operator                         │ 18.6%  │    6 │    0 │    0  │
│  PASS   │ 0.01s   │ github.com/cilium/cilium/operator/identity                │ 96.2%  │    1 │    0 │    0  │
│  PASS   │ 0.03s   │ github.com/cilium/cilium/operator/pkg/ciliumendpointslice │ 55.9%  │   23 │    0 │    0  │
│  PASS   │ 0.05s   │ github.com/cilium/cilium/operator/pkg/ingress             │ 27.1%  │   36 │    0 │    0  │
│  PASS   │ 4.11s   │ github.com/cilium/cilium/operator/watchers                │ 24.1%  │    2 │    0 │    0  │
│  PASS   │ 0.00s   │ github.com/cilium/cilium/pkg/addressing                   │ 51.6%  │    1 │    0 │    0  │
│  PASS   │ 0.04s   │ github.com/cilium/cilium/pkg/alibabacloud/eni             │ 57.7%  │    1 │    0 │    0  │
│  PASS   │ 0.00s   │ github.com/cilium/cilium/pkg/alibabacloud/utils           │ 88.9%  │    5 │    0 │    0  │
│  PASS   │ 0.01s   │ github.com/cilium/cilium/pkg/alignchecker                 │ 75.9%  │    1 │    0 │    0  │
│  PASS   │ 0.38s   │ github.com/cilium/cilium/pkg/allocator                    │ 63.7%  │    1 │    0 │    0  │
│  PASS   │ 1.20s   │ github.com/cilium/cilium/pkg/api/helpers                  │ 64.7%  │    1 │    0 │    0  │
│  PASS   │ 0.00s   │ github.com/cilium/cilium/pkg/api/metrics/mock             │ 100.0% │    1 │    0 │    0  │
│  PASS   │ 0.01s   │ github.com/cilium/cilium/pkg/aws/ec2                      │ 3.4%   │    8 │    0 │    0  │
│  PASS   │ 0.01s   │ github.com/cilium/cilium/pkg/aws/ec2/mock                 │ 39.4%  │    1 │    0 │    0  │
│  PASS   │ 6.30s   │ github.com/cilium/cilium/pkg/aws/eni                      │ 78.7%  │    1 │    0 │    0  │
│  PASS   │ 0.01s   │ github.com/cilium/cilium/pkg/aws/eni/limits               │ 61.5%  │    1 │    0 │    0  │
│  PASS   │ 0.01s   │ github.com/cilium/cilium/pkg/azure/api/mock               │ 84.3%  │    1 │    0 │    0  │
│  PASS   │ 0.14s   │ github.com/cilium/cilium/pkg/azure/ipam                   │ 76.8%  │    1 │    0 │    0  │
│  PASS   │ 0.00s   │ github.com/cilium/cilium/pkg/azure/types                  │ 18.0%  │    1 │    0 │    0  │
│  PASS   │ 0.01s   │ github.com/cilium/cilium/pkg/backoff                      │ 78.6%  │    1 │    0 │    0  │
│  PASS   │ 0.01s   │ github.com/cilium/cilium/pkg/bgp/config                   │ 85.7%  │    1 │    0 │    0  │
│  PASS   │ 0.01s   │ github.com/cilium/cilium/pkg/bgp/fence                    │ 38.1%  │    4 │    0 │    0  │
│  PASS   │ 0.03s   │ github.com/cilium/cilium/pkg/bgp/manager                  │ 32.4%  │    2 │    0 │    0  │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
make: *** [Makefile:200: integration-tests] Error 1
The command "./.travis/build.sh" exited with 2.
```

Cc @ldelossa @dylandreimerink

--- a/go.mod
+++ b/go.mod
@@ -229,7 +229,7 @@ replace (
 	github.com/miekg/dns => github.com/cilium/dns v1.1.4-0.20190417235132-8e25ec9a0ff3
 	github.com/optiopay/kafka => github.com/cilium/kafka v0.0.0-20180809090225-01ce283b732b
 
-	go.universe.tf/metallb => github.com/cilium/metallb v0.1.1-0.20220628131316-3c8101856f47
+	go.universe.tf/metallb => github.com/cilium/metallb v0.1.1-0.20220713202445-9066eee9e0be
 
 	// Using private fork of controller-tools. See commit msg for more context
 	// as to why we are using a private fork.

--- a/pkg/bgp/speaker/speaker_test.go
+++ b/pkg/bgp/speaker/speaker_test.go
@@ -351,9 +351,16 @@ func TestSpeakerOnUpdateNode(t *testing.T) {
 			atomic.AddInt32(&callCount, 1)
 			return types.SyncStateSuccess
 		},
-		PeerSession_: func() []metallbspr.Session {
+		GetBGPController_: func() *metallbspr.BGPController {
 			atomic.AddInt32(&callCount, 1)
-			return []metallbspr.Session{mockSession}
+			return &metallbspr.BGPController{
+				SvcAds: make(map[string][]*metallbbgp.Advertisement),
+				Peers: []*metallbspr.Peer{
+					{
+						BGP: mockSession,
+					},
+				},
+			}
 		},
 	}
 
@@ -449,6 +456,16 @@ func TestSpeakerOnDeleteNode(t *testing.T) {
 			atomic.AddInt32(&callCount, 1)
 			return []metallbspr.Session{mockSession}
 		},
+		GetBGPController_: func() *metallbspr.BGPController {
+			return &metallbspr.BGPController{
+				SvcAds: make(map[string][]*metallbbgp.Advertisement),
+				Peers: []*metallbspr.Peer{
+					{
+						BGP: mockSession,
+					},
+				},
+			}
+		},
 	}
 
 	spkr := &MetalLBSpeaker{
@@ -531,6 +548,16 @@ func TestSpeakerOnUpdateAndDeleteCiliumNode(t *testing.T) {
 		},
 		PeerSession_: func() []metallbspr.Session {
 			return []metallbspr.Session{mockSession}
+		},
+		GetBGPController_: func() *metallbspr.BGPController {
+			return &metallbspr.BGPController{
+				SvcAds: make(map[string][]*metallbbgp.Advertisement),
+				Peers: []*metallbspr.Peer{
+					{
+						BGP: mockSession,
+					},
+				},
+			}
 		},
 	}
 

